plugins/inputs/t128_graphql/query_builder.go

```diff
@@ -1,20 +1,162 @@
 package t128_graphql
 
-import "strings"
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
 
-func buildQuery(entryPoint string, fields map[string]string, tags map[string]string) string {
-	var replacer = strings.NewReplacer("[", "(", "]", "\")", "/", "{", ":", ":\"")
-	query := "query {" + replacer.Replace(entryPoint) + "{"
+	"github.com/Jeffail/gabs"
+)
 
-	for _, element := range fields {
-		query += "\n" + element
-	}
-	query = strings.TrimSpace(query)
-	for _, element := range tags {
-		query += "\n" + element
+const (
+	predicateTag = "$"
+)
+
+/*
+BuildQuery first creates an intermediary query object that is traversed by buildQueryBody() in pre-order
```

Comment:
Thanks for the example - it helped read and understand what your code is trying to do